common/config/rush/build-cache.json

```diff
@@ -0,0 +1,9 @@
+{
+  "buildCacheEnabled": true,
+  "cacheProvider": "azure-blob-storage",
+  "azureBlobStorageConfiguration": {
```

Comment:
I'd recommend we do a local-only cache first and get some data to make sure it's worth before we commit to doing azure blob

Reply:
That makes sense. Although it also fall back to local disk cache when cloud storage is not accessible.  With this included I hope we can test the cloud cache among a few team member first before we proceed to utilize CI.

Reply:
How about we update this PR to use `local-only` in `main`, and the team members who want to test cloud can edit this file locally to use a shared cloud storage?

Reply:
sure we can start with local-only. It will be a little inconvenient having to stash/apply the change everytime we make new branch.

Reply:
I don't want to make this too inconvenient for your team.  I'd be OK configuring cloud storage in main under the following conditions:

1. We require authentication even for read access
2. We create the storage account in our EngSys sub
3. We test `rush update-cloud-credentials`, understand how it works and that it is secure
4. We ensure the cache is local-only by default with no hassle to users, and users can opt-in (by running `rush update-cloud-credentials`)

Reply:
>We require authentication even for read access

Yes, we need a readonly SAS token to access blob cache.

> We create the storage account in our EngSys sub

Yes, definitely that's the right place.  I only have Reader access though so I am using playground for testing.

> We test rush update-cloud-credentials, understand how it works and that it is secure

There's two options for this command
1. passing in a valid SAS token via `--credential` option directly
2. using `--interactive` option. My understanding is that we sign in using our microsoft account and grant permission to AZ CLI, then rush uses AZ CLI to generate a readonly SAS token using our user account.  The user account also needs to be assigned `Data Reader` role to the Blob resource.

There's also the third way: setting the environment variable, which will overrides above.

> We ensure the cache is local-only by default with no hassle to users, and users can opt-in (by running rush update-cloud-credentials)

Without authentication, and with cloud cache configured in the `build-cache.json`, for the first clean build, user will see a warning of failed authentication, then rush would build and store result in local cache. later builds will get it from local cache

![image](https://github.com/Azure/azure-sdk-for-js/assets/7583839/0d24205f-d25f-41df-bfee-fd3acfaf8745)

Reply:
We can go with local for now. I already updated the PR.

Reply:
> There's two options for this command
> 
> 1. passing in a valid SAS token via `--credential` option directly
> 2. using `--interactive` option. My understanding is that we sign in using our microsoft account and grant permission to AZ CLI, then rush uses AZ CLI to generate a readonly SAS token using our user account.  The user account also needs to be assigned `Data Reader` role to the Blob resource.

With the changes to make playground auto-delete, I can create you an account in the engsys sub (when you are ready).  I think we should start with only using the `--credential` option (using a SAS stored in KeyVault acessible by the team) and not `--interactive` (which would require granting user permissions to the storage account).